test/integration/raft_test.lua

```diff
@@ -0,0 +1,474 @@
+                        replication_timeout: $replication_timeout
+                        synchro_timeout: $synchro_timeout
+                    ) {
+                        mode
+                        raft_quorum
+                        election_timeout
+                        replication_timeout
+                        synchro_timeout
+                    }
+                }
+            }
+        ]],
+        variables = vars,
+        raise = false,
+    })
+    if response.errors then
+        error(response.errors[1].message, 2)
+    end
+    return response.data.cluster.failover_params
+end
+
+g.before_all = function()
+    t.skip_if(box.ctl.on_election == nil)
+    g.cluster = h.Cluster:new({
+        datadir = fio.tempdir(),
+        use_vshard = true,
+        server_command = h.entrypoint('srv_raft'),
+        cookie = h.random_cookie(),
+        replicasets = {
+            {
+                alias = 'router',
+                uuid = h.uuid('a'),
+                roles = {
+                    'vshard-router',
+                    'test.roles.api',
+                },
+                servers = {
+                    {instance_uuid = h.uuid('a', 'a', 1)},
+                },
+            },
+            {
+                alias = 'storage',
+                uuid = replicaset_uuid,
+                roles = {
+                    'vshard-storage',
+                    'test.roles.storage',
+                },
+                servers = {
+                    {
+                        instance_uuid = storage_1_uuid,
+                    },
+                    {
+                        instance_uuid = storage_2_uuid,
+                    },
+                    {
+                        instance_uuid = storage_3_uuid,
+                        env = {
+                            TARANTOOL_ELECTION_MODE = 'voter',
+                        },
+                    },
+                },
+            },
+            {
+                alias = 'single-storage',
+                uuid = single_replicaset_uuid,
+                roles = {},
+                servers = {
+                    {
+                        instance_uuid = single_storage_uuid,
+                        env = {
+                            TARANTOOL_ELECTION_MODE = 'off',
+                        },
+                    },
+                },
+            },
+        },
+    })
+    g.cluster:start()
+
+    g.cluster.main_server:setup_replicaset({
+        roles = {'vshard-storage', 'test.roles.storage'},
+        weight = 0,
+        uuid = single_replicaset_uuid,
+    })
+    -- make single vshard-storage writable
+    g.cluster:server('single-storage-1'):exec(function()
+        box.cfg{replication_synchro_quorum = 1}
+    end)
+end
+
+g.after_all = function()
+    g.cluster:stop()
+    fio.rmtree(g.cluster.datadir)
+end
+
+local function set_master(instance_name)
+    g.cluster:server(instance_name):exec(function()
+        box.ctl.promote()
```

Comment:
I don't see any "box.ctl.promote" call in failover code. Should user call it manually?